### PR TITLE
feat(Authoring): Implement add lesson button menu next to lessons

### LINKED ISSUE #1671: feat(Authoring): Implement add lesson button menu next to lessons
Add an add lesson button next to all the active lessons in the project view. When the button is clicked, it should open a menu that lets the author "Add Lesson Before" or "Add Lesson After". When either of those are chosen, a new lesson should be added named "New Lesson" and it should have no steps.

### PR DESCRIPTION
**Pull request by geoffreykwan**
## Changes
- Added an add button next to all active lessons in the project view
- Clicking on the add button will open a menu that has the options "Add Lesson Before" and "Add Lesson After"
- Clicking on either of the options will create a new lesson named "New Lesson"
- The new lesson will have no steps

## Test
- Use the new add lesson button menus to add a lesson
  - before the first lesson
  - before a lesson that is not the first lesson
  - after a lesson
 - Add an unused lesson when there are no unused lessons
 - Add an active lesson when there are no active lessons

Closes #1671

### CODE REVIEW

**Review by hirokiterashima — COMMENTED** [review]
Functionality works. I had a few questions as I was reviewing:

1. Should we allow adding a new lesson to the unused lessons? We do this now in "Add new lesson" -> "Create your own" workflow.
2. Should the new functionality (addLessonBefore(), addLessonAfter(), and the template) be defined in a new AddNewLessonButtonComponent instead of in ProjectAuthoringComponent? It seems like the functions have nothing to do with the ProjectAuthoringComponent and therefore can be abstracted away into a separate class?

**Comment by geoffreykwan** [comment]
I've made these changes
- Created add lesson button component
- Added the ability to add unused lessons using the add lesson button menu
- Added the ability to add an unused lesson when there are no unused lessons
- Added the ability to add an active lesson when there are no active lessons

File: src/assets/wise5/authoringTool/add-lesson-button/add-lesson-button.component.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<button
+  mat-icon-button
+  color="primary"
+  class="add-lesson-button"
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/add-lesson-button/add-lesson-button.component.html]: Unused, remove?
```suggestion
```

File: src/assets/wise5/authoringTool/add-lesson-button/add-lesson-button.component.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  protected addLessonAfter(): void {
+    const newLesson = this.createNewLesson();
+    this.projectService.createNodeAfter(newLesson, this.lessonId);
+    this.updateProject(newLesson.id);
+  }
+
+  private createNewLesson(): any {
+    return this.projectService.createGroup('New Lesson');
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/add-lesson-button/add-lesson-button.component.ts]: Localize?

File: src/assets/wise5/authoringTool/project-authoring/project-authoring.component.html [hunk 3]
```diff
@@ -75,32 +75,51 @@
...
         (selectNodeEvent)="selectNode($event)"
         [showPosition]="false"
         [projectId]="projectId"
         [expanded]="lessonIdToExpanded()[inactiveGroupNode.id]"
         (onExpandedChanged)="onExpandedChanged($event)"
+        fxFlex
       ></project-authoring-lesson>
+      <add-lesson-button
+        class="add-lesson-button-next-to-lesson"
+        [active]="false"
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/project-authoring/project-authoring.component.html]: Unnecessary, remove?
```suggestion
```

File: src/assets/wise5/authoringTool/project-authoring/project-authoring.component.html [hunk 4]
```diff
@@ -75,32 +75,51 @@
...
+      [lessonId]="lesson.id"
+    ></add-lesson-button>
   </div>
   <div>
     <h6 class="unused-header" i18n>Unused Lessons</h6>
-    <div *ngIf="inactiveGroupNodes.length === 0" i18n>There are no Unused Lessons</div>
-    <div *ngFor="let inactiveGroupNode of inactiveGroupNodes">
+    <div *ngIf="inactiveGroupNodes.length === 0" fxLayout="row" fxLayoutAlign="start center">
+      <div i18n>There are no unused lessons</div>
+      <add-lesson-button [active]="false"></add-lesson-button>
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/project-authoring/project-authoring.component.html]: Unnecessary attribute, remove?
```suggestion
      <add-lesson-button></add-lesson-button>
```

**Review by hirokiterashima — APPROVED** [review]
LGTM. Can you address these before merge?

- Adding a lesson before/after in the active section doesn't highlight the newly-added lesson. Highlighting new lessons seems to work in the inactive section.
- I added inline comments about code.

**Pull request merged**